src/hostapi/wdmks/pa_win_wdmks.c

```diff
@@ -163,8 +163,8 @@ Default is to use the pin category.
 #define DYNAMIC_GUID(data) {data}
```

Comment:
Look on line 161. probably a better fix is to change that

'''#ifdef _MSC_VER and !defined(__clang__) // MSC but not clang-cl
'''

Reply:
From what I tested, `#define DEFINE_GUID(n,data) EXTERN_C const GUID n = {data}` works in MSVC cl (Visual Studio Community). It also works when building with vcpkg. My version works on both MSVC cl and Clang cl but I don't have mingw. 
Another fix would be to let Windows headers build static GUID instead of external ones:
```
#define INITGUID
#define DYNAMIC_GUID(data) {data}
```

Reply:
Do you mean something like this ? 
```
#ifdef _MSC_VER
#define NOMMIDS
#define DYNAMIC_GUID(data) {data}
#define _NTRTL_ /* Turn off default definition of DEFINE_GUIDEX */
#undef DEFINE_GUID
#ifndef __clang__
  #define DEFINE_GUID(n, data) EXTERN_C const GUID n = {data}
  #define DEFINE_GUID_THUNK(n, data) DEFINE_GUID(n, data)
  #define DEFINE_GUIDEX(n) DEFINE_GUID_THUNK(n, STATIC_##n)
#else
  #define DEFINE_GUID(n, ...) EXTERN_C const GUID n = {__VA_ARGS__}
  #define DEFINE_GUID_THUNK(n, ...) DEFINE_GUID(n, __VA_ARGS__)
  #define DEFINE_GUIDEX(n) DEFINE_GUID_THUNK(n, STATIC_##n)
  #endif
#endif
```

Reply:
Why not just a simple:
```
#ifdef _MSC_VER
#define NOMMIDS
#define DYNAMIC_GUID(data) {data}
#define INIT_GUID
#define DEFINE_GUIDEX(n) EXTERN_C const GUID n = {STATIC_##n}
#endif
```

?

Reply:
No, I do not mean what you suggested, I mean the one line change: 

    #ifdef _MSC_VER && !defined( __clang__ )
    #define NOMMIDS	#define NOMMIDS
    #define DYNAMIC_GUID(data) {data}	#define DYNAMIC_GUID(data) {data}
    ...

Reply:
The code block that you propose to change only applies to MSVC. For the GCC (and clang, and clang-cl) there is a definition of `DEFINE_GUID_THUNK` further up in the file.

Reply:
> 
> 
> No, I do not mean what you suggested, I mean the one line change:
> 
> ```
> #ifdef _MSC_VER && !defined( __clang__ )
> #define NOMMIDS	#define NOMMIDS
> #define DYNAMIC_GUID(data) {data}	#define DYNAMIC_GUID(data) {data}
> ...
> ```

This has the same effect as if you would comment out the block on MSVC, DYNAMIC_GUID is not defined and STATIC_## macros are not used:
```
src/hostapi/wdmks/pa_win_wdmks.c(2215,10): error: initializing 'GUID' (aka 'struct _GUID') with an expression of incompatible type 'int'
    GUID guid = DYNAMIC_GUID( DEFINE_WAVEFORMATEX_GUID(format->wFormatTag) );
         ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
@Nemirtingas ok, if the following works, could we do it exactly like the following please?

```
#ifdef _MSC_VER
#define NOMMIDS
#define DYNAMIC_GUID(data) {data}
#define _NTRTL_ /* Turn off default definition of DEFINE_GUIDEX */
#undef DEFINE_GUID
#ifdef __clang__ /* clang-cl: avoid too many arguments error */
  #define DEFINE_GUID(n, ...) EXTERN_C const GUID n = {__VA_ARGS__}
  #define DEFINE_GUID_THUNK(n, ...) DEFINE_GUID(n, __VA_ARGS__)
  #define DEFINE_GUIDEX(n) DEFINE_GUID_THUNK(n, STATIC_##n)
#else
  #define DEFINE_GUID(n, data) EXTERN_C const GUID n = {data}
  #define DEFINE_GUID_THUNK(n, data) DEFINE_GUID(n, data)
  #define DEFINE_GUIDEX(n) DEFINE_GUID_THUNK(n, STATIC_##n)
#endif /* __clang__ */
#endif
```

Reply:
I'm ok with this.

Reply:
@Nemirtingas could you update the pull request with this agreed version please.